### PR TITLE
added more api gateway coverage

### PR DESCRIPTION
**Pull request by tootedom**
Hi there,

Firstly, thank you very much for all the work on moto.  Great library.

I've added extra support for apigateway, as a result of developing ansible modules using boto3 from provisioning api gateways.  This pull request covers the features we are intending to use at work, so I built the additional mocking around the testing of our ansible modules.

The features added are:

- get_stages, 
- stage not found exception 
- update stage configuration
- descriptions on deployments
- creation date on deployments
- setting stage variables on deployments and stage creation

Hopefully the changes make sense.  Please let me know if you need us to make any changes.
thanks
/dom

### CODE REVIEW

File: moto/apigateway/models.py [hunk 1]
```diff
@@ -150,24 +153,124 @@ def delete_integration(self, method_type):
 
 
 class Stage(dict):
-    def __init__(self, name=None, deployment_id=None):
+
+
+    def __init__(self, name=None, deployment_id=None, variables={}):
```
Comment (spulec) [inline moto/apigateway/models.py]: Generally not a good idea to have mutable things as default arguments. See [here](http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments) for why. It is one of the bigger gotchas in python IMO. It looks like you have a similar thing in a couple other places too.

**Comment by spulec** [comment]
This is great!

I had one comment about mutable things as default arguments, but would be happy to merge if we can get that cleaned up.

**Comment by tootedom** [comment]
>>
>> Generally not a good idea to have mutable things as default arguments
>>

You learn something new everyday.  Thanks for that.  
Think I've updated them all.

cheers
/dom

**Comment by tootedom** [comment]
Hi there,

I just noticed a bug in the former commit that was always setting 'cacheClusterSize'.  It appears this does not come back on a newly created stage, until the caching is enabled.  So I've catered for this by setting 'cacheClusterSize' to 0.5 if cacheClusterEnabled is set to true (and it's not previously defined).  At the same time the cache cluster size is a string (no idea why - seems that way returned from apigw).

I've also whilst fixing the above edge case.  Extended create stage to allow for taking:

- description, cacheClusterEnabled and cacheClusterSize.


Sorry for the update.  Only, just noticed this edge case this morning.

/dom

**Comment by tootedom** [comment]
Hi there,

I'm also a moron, and typo'd the NotFoundException as NonFoundException.  I've fixed this.

Apologies
/dom

**Comment by spulec** [comment]
Perfect, thanks!

**Pull request merged**